intelliserver/middleware/awsConfigProvider.js

```diff
@@ -0,0 +1,21 @@
+// awsConfigMiddleware.js
+const AWS = require('aws-sdk');
+const config = require('../config');
+
+const awsConfigProvider = (req, res, next) => {
+    const accessKeyId = req.header('X-aws-access-Key');
+    const secretAccessKey = req.header('X-aws-secret-Key');
+    const region = req.header('X-aws-region');
+
+    if (accessKeyId && secretAccessKey) {
+        // Set up AWS SDK configuration with user-provided access key and secret
+        AWS.config.update({
```

Comment:
you updating a global variable based on received header, this will leak the keys between users.

Create the key provider similar to the other functions inside intelliServer (the same way done for openai or chatbot).

Reply:
I couldn't find any API based instantiation of AWS service and the same with Google Cloud SDK (it has vision API that will take the API key and then return the data)

Reply:
In the latest commit, I have added conditional switching between provider so that can fix this issue, can you check if this will work?